cdap-app-fabric/src/test/java/co/cask/cdap/metadata/NoopFieldLineageReader.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.metadata;
+
+import co.cask.cdap.api.lineage.field.EndPoint;
+import co.cask.cdap.api.lineage.field.InputField;
+import co.cask.cdap.api.lineage.field.Operation;
+import co.cask.cdap.api.lineage.field.ReadOperation;
+import co.cask.cdap.api.lineage.field.TransformOperation;
+import co.cask.cdap.api.lineage.field.WriteOperation;
+import co.cask.cdap.common.app.RunIds;
+import co.cask.cdap.data2.metadata.lineage.field.EndPointField;
+import co.cask.cdap.data2.metadata.lineage.field.FieldLineageReader;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.cdap.proto.id.ProgramId;
+import co.cask.cdap.proto.id.ProgramRunId;
+import co.cask.cdap.proto.metadata.lineage.ProgramRunOperations;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Mock implementation of the {@link FieldLineageReader} for testing purpose.
+ */
+public class NoopFieldLineageReader implements FieldLineageReader {
```

Comment:
Let's call this a FakeFieldLineageReader. I would expect a no-op to return empty everything.

Reply:
renamed